algebird-test/src/test/scala/com/twitter/algebird/macros/CaseClassMacrosTest.scala

```diff
@@ -0,0 +1,45 @@
+package com.twitter.algebird.macros
+
+import com.twitter.algebird._
+import com.twitter.algebird.macros.caseclass._
+import com.twitter.algebird.macros.ArbitraryCaseClassMacro.arbitrary
+
+import org.scalatest._
+import org.scalacheck.Arbitrary
+import org.scalatest.prop.{ GeneratorDrivenPropertyChecks, PropertyChecks }
+
+class CaseClassMacrosTest extends PropSpec with PropertyChecks with Matchers {
+  import BaseProperties._
+
+  implicit val arbitraryFoo: Arbitrary[Foo] = arbitrary[Foo]
+  implicit val arbitraryBar: Arbitrary[Bar] = arbitrary[Bar]
+
+  case class Foo(a: Int, b: Short, c: Long)
+  case class Bar(a: Boolean, foo: Foo)
+
+  property("Foo is a Semigroup") {
```

Comment:
I'm not sure these are running. We had some issue with this in the past. I really hate scalatest. It has so many styles.

The risk is this: semigroupLaws returns a Prop. It does not do any asserts. But scalatest works on asserts. So we had tests that were failing, but we didn't notice for a long time due to this.

Rather than really fix the tests, we make a trait to do the right thing, and mixed that in. The problem is, copying that is easy to forget or hard to review singe you need to remember this issue.

I think this might not be reporting errors since I think it is expecting an assert inside the property, block.

Can we actually not use scalatest here and just use scalacheck? It is too easy to get these tests wrong.